### ISSUE TITLE
No Error message

### ISSUE THREAD

**Issue by Skeeve** [SIWECOS/HSHS-DOMXSS-Scanner#15]
Currently the core api does not receive an error when your scanner runs into issues like the domain is not reachable.

Currently we handle this with a timeout, but this is just a workaround.

Please provide, via callback, an error message to the core api.

An example could be (according to the api):

```JSON
[{
  "name": "ERROR",
  "hasError": true,
  "dangerlevel": 0,
  "errorMessage": {
    "placeholder": "NO_RESPONSE",
    "values": {
      "domain": "http://your.do.main",
    }
  },
  "score": 0,
  "scoreType": "success",
  "testDetails": []
}]
```

Check https://github.com/SIWECOS/siwecos-core-api/issues/68#issuecomment-378510683 for more information, especially about example values for the ERROR and NO_RESPONSE placeholders.

Please make sure to use error placeholders which are either specific for your scanner (like DOMXSS_OUT_OF_MEMORY) if it is a scanner specific error message or, if the error is a general one which can happen to any scanner, non-specific (like NO_RESPONSE).

**Comment by Skeeve** [comment]
Looking into the log I see a proper reply. Issue lies in the core api.

Sorry: Until yesterday I was not aware of where to see the logs. Missing documentation :(

**Comment by Weegy** [comment]
public static function getTimeOutArray( string $scanner ) {
		$timeout                                           = array();
		$timeout['name']                                   = 'TIMEOUT';
		$timeout['hasError']                               = true;
		$timeout['dangerlevel']                            = 0;
		$timeout['score']                                  = 0;
		$timeout['scoreType']                              = 'success';
		$timeout['testDetails']                            = array();
		$timeout['errorMessage']                           = array();
		$timeout['errorMessage']['placeholder']            = 'SCANNER_TIMEOUT';
		$timeout['errorMessage']['values']                 = array();
		$timeout['errorMessage']['values']['scanner']      = $scanner;
		$timeout['errorMessage']['values']['timeoutvalue'] = 300;
		return json_encode($timeout);

	}

**Comment by Skeeve** [comment]
Please see @Weegy's comment above. Your error messgae currently looks like this:

```JSON
{
  "name":"DOMXSS",
  "hasError":true,
  "errorMessage":"NO_HTTP_RESPONSE",
  "score":0,
  "tests":[]
}
```

If you check the API you can see that errorMessage should be a "#/components/schemas/TranslateableMessage", not a string.

**Comment by Lednerb** [comment]
Closed by c99e397e3bb079e8c55cc1f4d06cf5e809be3127 and https://github.com/SIWECOS/HSHS-DOMXSS-Scanner/commit/755309454b7163c1582e2095b6413e39640a72ae.
